src/core/storage/fileio/s3_api.cpp

```diff
@@ -111,20 +114,57 @@ bool bucket_name_valid(const std::string& bucket_name) {
 }
 
 
-std::string string_from_s3url(const s3url& parsed_url) {
-  std::string ret = "s3://" + parsed_url.access_key_id + ":"
-      + parsed_url.secret_key + ":";
+} // anonymous namespace
+
+S3Client init_aws_sdk_with_turi_env(s3url& parsed_url) {
```

Comment:
Is it expected to change the parsed_url struct and pass those changes back?  If it's not expected to pass the changes back, then make it pass-by-value; otherwise add a comment here about how it will change.

Reply:
yes, it will change the optional sections of s3url, such as region or endpoint.